server/src/main/java/org/elasticsearch/index/query/PercolatorExecutionContext.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperBuilderContext;
+import org.elasticsearch.index.mapper.TextFieldMapper;
+import org.elasticsearch.index.query.support.NestedScope;
+
+public class PercolatorExecutionContext extends SearchExecutionContext {
+
+    private boolean allowUnmappedFields;
+    private final boolean mapUnmappedFieldAsString;
+
+    public PercolatorExecutionContext(
+        final SearchExecutionContext searchExecutionContext,
+        boolean allowUnmappedFields,
```

Comment:
I think `allowUnmappedFields` should always be `false`?  The whole point of the percolator is that you need to know mappings up front so that we can build the query index.

Reply:
I left it as a parameter because it is set again in `reset()` using `indexSettings.isDefaultAllowUnmappedFields()` which might be `true` right?